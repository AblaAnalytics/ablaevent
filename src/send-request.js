export const encodePostDataBody = (data) => {
    let body_data
    if (Array.isArray(data)) {
        body_data = 'data=' + encodeURIComponent(data)
    } else {
        body_data = 'data=' + encodeURIComponent(data['data'])
    }
    // delete data['data']

    if (data['compression']) {
        body_data += '&compression=' + data['compression']
        // delete data['compression']
    }

    return body_data
}
