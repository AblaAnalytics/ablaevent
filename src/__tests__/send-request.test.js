import { formEncodePostData } from '../send-request'

describe('formEncodePostData()', () => {
    given('subject', () => formEncodePostData(given.data))

    it('handles arrays', () => {
        given('data', () => ['foo', 'bar'])

        expect(given.subject).toMatchSnapshot()
    })

    it('handles objects', () => {
        given('data', () => ({ data: 'content' }))

        expect(given.subject).toMatchSnapshot()
    })

    it('handles data with compression', () => {
        given('data', () => ({ data: 'content', compression: 'lz64' }))

        expect(given.subject).toMatchSnapshot()
    })
})
