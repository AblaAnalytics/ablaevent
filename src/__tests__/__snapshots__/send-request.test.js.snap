// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encodePostDataBody() handles arrays 1`] = `"data=foo%2Cbar"`;

exports[`encodePostDataBody() handles data with compression 1`] = `"data=content&compression=lz64"`;

exports[`encodePostDataBody() handles objects 1`] = `"data=content"`;
